lib/syncUserFollowers.js

```diff
@@ -0,0 +1,124 @@
+				// get the concepts (topics) followed by the user
+				return myFT.getConceptsFollowedByUser(userId)
+					.catch(err => {
+						// if no concepts are found
+						if (err instanceof clientErrors.NotFoundError) {
+							return [];
+						}
+						throw err;
+					})
+					.then(conceptsResp => {
+						if (Array.isArray(conceptsResp)) {
+							log.silly({operation, subOp: 'getConceptsFollowedByUser', userId, groupId, conceptRes: JSON.stringify(conceptsResp)});
+							const userConceptIds = conceptsResp.map(concept => concept.uuid);
+							log.silly({operation, subOp: 'userConceptsFollowed', userId, groupId, userConceptsCount: userConceptIds.length});
+							// get the new concepts to be followed
+							const newConceptsToFollow = groupConcepts.filter((item)=> userConceptIds.indexOf(item.uuid) === -1);
+							if (newConceptsToFollow.length === 0) {
+								log.silly({operation, subOp: 'noNewConceptsToFollow', userId, groupId});
+								return {user: {uuid: userId, group: groupId, status: 'synchronisationIgnored', reason: 'noNewConceptsToFollow'}};
+							}
+
+							log.silly({operation, subOp: 'newConceptsToFollow', userId, group: groupId, newConceptsToFollow: JSON.stringify(newConceptsToFollow)});
+							const followProps = Object.assign({}, followedProperties);
+							followProps.asMemberOf = groupId;
+							// set the user as being a follower on the new concepts
+							return myFT.addConceptsFollowedByKatUser(userId, newConceptsToFollow, followProps)
+								.then(() => {
+									//Not sure we need to be doing this
+									log.silly({operation, subOp: 'getEmailDigestPreference', userId});
+									return myFT.getEmailDigestPreference(userId)
+										.catch(err => {
+											// if no preferences are found
+											if (err instanceof clientErrors.NotFoundError) {
+												log.silly({operation, subOp: 'getEmailDigestPreference', userId});
+												return myFT.setEmailDigestPreference(userId, digestProperties);
+											}
+											throw err;
+										});
+								})
+								.then(() => {
+									const cleanConcepts = newConceptsToFollow.map(item => {
+										const newItem = Object.assign({}, item);
+										delete newItem._rel;
+										return newItem;
+									});
+
+									log.silly({operation, subOp: 'kinesis.write', userId});
+									return kinesis.write(userId, 'subscribe', cleanConcepts);
+								})
+								.then((res) => {
+									log.silly({operation, subOp: 'kinesis res', kineisRes: JSON.stringify(res)});
+								})
+								.then(()=> ({user: {uuid: userId, status: 'synchronisationCompleted', group: groupId, newConceptsToFollow}}));
+						}
+						const msg = 'User conceptsResp is not an array';
+						log.error({operation, conceptsResp, status: msg});
+						throw new Error(msg);
+					});
+
+			}
+			log.silly({operation, subOp: 'noGroupConceptsToFollow', userId, groupId});
```

Comment:
There are loads of `silly` logs in this file. do we need all of them to be separate? And/or silly?